src/rtcp/ByePacket.ts

```diff
@@ -0,0 +1,290 @@
+	 *   packet (with just the minimal fixed header) will be created.
+	 *
+	 * @throws
+	 * - If given `view` does not contain a valid RTCP Bye packet.
+	 */
+	constructor(view?: DataView)
+	{
+		super(RtcpPacketType.BYE, view);
+
+		if (!view)
+		{
+			this.packetView = new DataView(new ArrayBuffer(FIXED_HEADER_LENGTH));
+
+			// Write version and packet type.
+			this.writeCommonHeader();
+
+			return;
+		}
+
+		if (getRtcpPacketType(view) !== RtcpPacketType.BYE)
+		{
+			throw new TypeError('not a RTCP Bye packet');
+		}
+		else if (getRtcpLength(view) !== view.byteLength)
+		{
+			throw new RangeError(
+				`length in the RTCP header (${getRtcpLength(view)} bytes) does not match view length (${view.byteLength} bytes)`
+			);
+		}
+
+		this.packetView = view;
+
+		// Position relative to the DataView byte offset.
+		let pos = 0;
+
+		// Move to SSRC/CSRC field(s).
+		pos += FIXED_HEADER_LENGTH;
+
+		let count = this.getCount();
+
+		while (count-- > 0)
+		{
+			const ssrc = this.packetView.getUint32(pos);
+
+			this.#ssrcs.push(ssrc);
+
+			pos += 4;
+		}
+
+		// Get padding.
+		if (this.getPaddingBit())
+		{
+			this.padding = this.packetView.getUint8(this.packetView.byteLength - 1);
+		}
+
+		// Check if there is reason.
+		if (pos + this.padding < this.packetView.byteLength)
+		{
+			const reasonLength = this.packetView.getUint8(pos);
+			const reasonPadding = -(reasonLength + 1) & 3;
+
+			// Move to the reason field.
+			pos += 1;
+
+			if (
+				pos + reasonLength + reasonPadding + this.padding !==
+				this.packetView.byteLength
+			)
+			{
+				throw new TypeError('invalid RTCP Bye packet');
+			}
+
+			const reasonView = new DataView(
+				this.packetView.buffer,
+				this.packetView.byteOffset + pos,
+				reasonLength
+			);
+
+			this.#reason = dataViewToString(reasonView);
+
+			pos += reasonLength + reasonPadding;
+		}
+
+		pos += this.padding;
+
+		// Ensure that view length and parsed length match.
+		if (pos !== this.packetView.byteLength)
+		{
+			throw new RangeError(
+				`parsed length (${pos} bytes) does not match view length (${this.packetView.byteLength} bytes)`
+			);
+		}
+	}
+
+	/**
+	 * Dump RTCP Bye packet info.
+	 */
+	dump(): ByePacketDump
+	{
+		return {
+			...super.dump(),
+			ssrcs  : this.#ssrcs,
+			reason : this.#reason
+		};
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	getByteLength(): number
+	{
+		let packetLength = FIXED_HEADER_LENGTH + (this.#ssrcs.length * 4);
+
+		if (this.#reason)
+		{
+			const reasonLength = getStringByteLength(this.#reason);
+			const reasonPadding = -(reasonLength + 1) & 3;
+
+			packetLength += 1 + reasonLength + reasonPadding;
+		}
+
+		packetLength += this.padding;
+
+		return packetLength;
+	}
+
+	/**
+	 * Get SSRC values.
+	 */
+	getSsrcs(): number[]
+	{
+		return Array.from(this.#ssrcs);
+	}
+
+	/**
+	 * Set SSRC values.
+	 *
+	 * @remarks
+	 * - Serialization is needed after calling this method.
+	 */
+	setSsrcs(ssrcs: number[]): void
+	{
+		this.#ssrcs = ssrcs;
```

Comment:
For the same reason we return a copy of `this.#ssrcs` in `getSsrcs()``

Otherwise we are keeping a reference to the provided `ssrcs` array.

```suggestion
		this.#ssrcs = Array,.from(ssrcs);
```

Reply:
Done here and everywhere.